### PR TITLE
adds contributing guidelines

### PR DESCRIPTION
**Pull request by JBAhire**
This PR,
- adds contributing guidelines in `.github` folder.

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
+# Contributing
+ 
+When contributing to this repository, please first discuss the change you wish to make via issue,
+or [Hypertrace slack](https://bit.ly/hypertrace-slack) with the owners of this repository before making a change.
```
Comment (pavolloffay) [inline .github/CONTRIBUTING.md:4]: I hope this won't expire. Full link might be better.

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+  an existing issue) or describe the issue + your fix (if submitting a wholly
+  new bugfix).
+4. Hit 'submit'! And please be patient - the maintainers will get to you when
+  they can.
+ 
+## Support Channels
+---
+Whether you are a user or contributor, official support channels include:
+- GitHub issues: https://github.com/hypertrace/hypertrace/issues/new
+- Slack: `#help` channel in [Hypertrace slack](https://bit.ly/hypertrace-slack)
```
Comment (pavolloffay) [inline .github/CONTRIBUTING.md:134]: The same here about the link expiration

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+     rolling back a few minor versions (e.g. if on 1.7, try 1.5 or 1.6) and
+     see if the problem goes away. This will help the devs narrow down when
+     the problem first arose in the commit log.
+   * **Try switching up dependency versions.** If the software in question has
+     dependencies (other libraries, etc) try upgrading/downgrading those as
+     well.
+ 
+* **Search the GitHub issues** to make sure it's not a known
+ issue.
+* If you don't find a pre-existing issue, consider **checking with the slack channel** in case the problem is non-bug-related.
```
Comment (jcchavezs) [inline .github/CONTRIBUTING.md:30]: Not 100% sure about this. Search in slack could be hard.

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+* Perform **basic troubleshooting** steps:
+ 
+   * **Make sure you're on the latest version.** If you're not on the most
+     recent version, your problem may have been solved already! Upgrading is
+     always the best first step.
+   * **Try older versions.** If you're already *on* the latest release, try
+     rolling back a few minor versions (e.g. if on 1.7, try 1.5 or 1.6) and
+     see if the problem goes away. This will help the devs narrow down when
+     the problem first arose in the commit log.
+   * **Try switching up dependency versions.** If the software in question has
```
Comment (jcchavezs) [inline .github/CONTRIBUTING.md:24]: Does this apply to this repo?

File: .github/CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+-------------
+ 
+Before submitting a bug, please do the following:
+ 
+* Perform **basic troubleshooting** steps:
+ 
+   * **Make sure you're on the latest version.** If you're not on the most
+     recent version, your problem may have been solved already! Upgrading is
+     always the best first step.
+   * **Try older versions.** If you're already *on* the latest release, try
```
Comment (jcchavezs) [inline .github/CONTRIBUTING.md:20]: We don't have versions yet.

File: .github/CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+* If you don't find a pre-existing issue, consider **checking with the slack channel** in case the problem is non-bug-related.
+ 
+### What to put in your issue
+------------------------------
+ 
+Make sure issue you raised gets the attention it deserves: issues with missing
+information may be ignored or punted back to you, delaying a fix.  The below
+constitutes a bare minimum; more info is almost always better:
+ 
+* **What version of the core programming language interpreter/compiler are you
```
Comment (jcchavezs) [inline .github/CONTRIBUTING.md:39]: Does this apply?

File: .github/CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+ using?** For example, Most of our services are written in Java so you what version of Java you're on?
+* **What operating system are you on?** Windows? (10? Home? Pro?)
+ Mac OS X?  (10.x.x?) Linux? (Which distro? Which version of that
+ distro? 32 or 64 bits?) Again, more detail is better.
+* **Are you using latest codebase and are on main branch?** Ideally, you
+ followed the advice above and have ruled out (or verified that the problem
+ exists in) a few different versions.
+* **How can the developers recreate the bug on their end?** If possible,
```
Comment (jcchavezs) [inline .github/CONTRIBUTING.md:47]: If possible, include a copy of your code *or an example repo*, the command you used to invoke it, and the full
 output of your run (if applicable.)

File: .github/CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,142 @@
+# Contributing
+
```
Comment (pavolloffay) [inline .github/CONTRIBUTING.md:2]: I would include some intro that we are OSS project, maybe with used licenses explanation.

Then a couple of words about in lines like: "'We like contributions, please help us make a project better..."

File: .github/CONTRIBUTING.md (hunk 3 above)
Comment (JBAhire) [reply .github/CONTRIBUTING.md:30]: this was more in a context of they can check with other people on slack if this is known issue.

File: .github/CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+     the problem first arose in the commit log.
+   * **Try switching up dependency versions.** If the software in question has
+     dependencies (other libraries, etc) try upgrading/downgrading those as
+     well.
+ 
+* **Search the GitHub issues** to make sure it's not a known
+ issue.
+* If you don't find a pre-existing issue, consider **checking with the slack channel** in case the problem is non-bug-related.
+ 
+### What to put in your issue
```
Comment (pavolloffay) [inline .github/CONTRIBUTING.md:32]: We probably don't need this since we have or will have issue templates.

Also the bullet points do not seem right, most folks will consumer only docker images so probably we need info about the docker/orchestrator.

File: .github/CONTRIBUTING.md (hunk 4 above)
Comment (JBAhire) [reply .github/CONTRIBUTING.md]: It does in case for Helm.

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (JBAhire) [reply .github/CONTRIBUTING.md:5]: This will expire but we will update from bit.ly dashboard.

File: .github/CONTRIBUTING.md (hunk 5 above)
Comment (JBAhire) [reply .github/CONTRIBUTING.md]: yup. removing.

File: .github/CONTRIBUTING.md (hunk 9 above)
Comment (JBAhire) [reply .github/CONTRIBUTING.md:22]: makes sense. addressed.

File: .github/CONTRIBUTING.md (hunk 7 above)
Comment (JBAhire) [reply .github/CONTRIBUTING.md:37]: should we completely remove this section and add this things in issue template?

Comment (JBAhire) [reply .github/CONTRIBUTING.md:37]: done!

File: .github/CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+ 
+Any bugfix that doesn't include a test proving the existence of the bug being
+fixed, may be suspect.  Ditto for new features that can't prove they actually
+work.
+ 
+We've found that test-first development really helps make features better
+architected and identifies potential edge cases earlier instead of later.
+Writing tests before the implementation is strongly encouraged.
+ 
+## Full example
```
Comment (pavolloffay) [inline .github/CONTRIBUTING.md:84]: I am not sure I like this title.

e.g. Example workflow for submitting a change?

**Review by pavolloffay — APPROVED** [review]
LGTM. We can always improve later.
